var/spack/repos/builtin/packages/flux/package.py

```diff
@@ -45,20 +46,30 @@ class Flux(Package):
     depends_on("libxslt")
     depends_on("python")
     depends_on("py-cffi")
+    depends_on("jansson")
 
     # TODO: This provides a catalog, hacked with environment below for now
-    depends_on("docbook-xml", type='build')
-    depends_on("asciidoc", type='build')
+    # depends_on("docbook-xml", type='build')
+    # depends_on("asciidoc", type='build')
 
-    def install(self, spec, prefix):
-        # Bootstrap with autotools
-        bash = which('bash')
-        bash('./autogen.sh')
-        bash('./autogen.sh')  # yes, twice, intentionally
+    depends_on("autoconf", type='build')
+    depends_on("automake", type='build')
+    depends_on("libtool", type='build')
+
+    def autoreconf(self, spec, prefix):
+        if os.path.exists('autogen.sh'):
+            # Bootstrap with autotools
+            bash = which('bash')
+            bash('./autogen.sh')
+            bash('./autogen.sh')  # yes, twice, intentionally
```

Comment:
Yeah, it's an issue with the order of operations, there's a kind of cyclic dependency between aclocal and libtools with some autotools versions because of the way everything is specified.  This looks odd but makes it version independent (within reason).